src/components/svgs/nodes/chengdumetro-line-badge.tsx

```diff
@@ -0,0 +1,181 @@
+import { MonoColour } from '@railmapgen/rmg-palette-resources';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { RmgFields, RmgFieldsField } from '@railmapgen/rmg-components';
+import { AttrsProps, CityCode } from '../../../constants/constants';
+import { MiscNodeType, Node, NodeComponentProps } from '../../../constants/nodes';
+import { AttributesWithColor, ColorField } from '../../panels/details/color-field';
+import { MultilineText } from '../common/multiline-text';
+
+const ChengduMetroLineBadge = (props: NodeComponentProps<ChengduMetroLineBadgeAttributes>) => {
+    const { id, x, y, attrs, handlePointerDown, handlePointerMove, handlePointerUp } = props;
+    const {
+        num = defaultChengduMetroLineBadgeAttributes.num,
+        color = defaultChengduMetroLineBadgeAttributes.color,
+        type = defaultChengduMetroLineBadgeAttributes.type,
+    } = attrs ?? defaultChengduMetroLineBadgeAttributes;
+
+    const onPointerDown = React.useCallback(
+        (e: React.PointerEvent<SVGElement>) => handlePointerDown(id, e),
+        [id, handlePointerDown]
+    );
+    const onPointerMove = React.useCallback(
+        (e: React.PointerEvent<SVGElement>) => handlePointerMove(id, e),
+        [id, handlePointerMove]
+    );
+    const onPointerUp = React.useCallback(
+        (e: React.PointerEvent<SVGElement>) => handlePointerUp(id, e),
+        [id, handlePointerUp]
+    );
+
+    const fgColor = color[3];
+    const fontSize = 16;
+
+    return (
+        <g
+            id={id}
```

Comment:
You must have an overlay transparent element to have this id ```id={`stn_core_${id}`}``` otherwise when releasing the pointer on this node, it will not be connectable.